library/Reporting/Report.php

```diff
@@ -37,91 +37,39 @@ class Report
     protected $template;
 
     /**
-     * @param int $id
+     * Transforms the report model to this class
      *
-     * @return  static
+     * @param Model\Report $reportModel
      *
-     * @throws  Exception
+     * @return Report
+     * @throws Exception
      */
-    public static function fromDb($id)
+    public static function fromModel(Model\Report $reportModel): Report
     {
         $report = new static();
 
-        $db = $report->getDb();
-
-        $select = (new Sql\Select())
-            ->from('report')
-            ->columns('*')
-            ->where(['id = ?' => $id]);
-
-        $row = $db->select($select)->fetch();
-
-        if ($row === false) {
-            throw new Exception('Report not found');
-        }
+        $report->id = $reportModel->id;
+        $report->name = $reportModel->name;
+        $report->author = $reportModel->author;
+        $report->timeframe = Timeframe::fromModel($reportModel->timeframe);
+        $report->template = Template::fromModel($reportModel->template);
 
-        $report
-            ->setId($row->id)
-            ->setName($row->name)
-            ->setAuthor($row->author)
-            ->setTimeframe(Timeframe::fromDb($row->timeframe_id))
-            ->setTemplate(Template::fromDb($row->template_id));
-
-        $select = (new Sql\Select())
-            ->from('reportlet')
-            ->columns('*')
-            ->where(['report_id = ?' => $id]);
-
-        $row = $db->select($select)->fetch();
-
-        if ($row === false) {
+        if ($reportModel->reportlets === null) {
             throw new Exception('No reportlets configured.');
         }
```

Comment:
Please do this check after the below foreach loop and check if `! empty($reportlets)`.